src/frontend/src/optimizer/plan_node/logical_agg.rs

```diff
@@ -1253,29 +1284,26 @@ impl PredicatePushdown for LogicalAgg {
 
 impl ToBatch for LogicalAgg {
     fn to_batch(&self) -> Result<PlanRef> {
+        self.to_batch_with_order_required(&Order::any())
+    }
+
+    fn to_batch_with_order_required(&self, required_order: &Order) -> Result<PlanRef> {
         let new_input = self.input().to_batch()?;
```

Comment:
maybe we can check the required order at first and after we decide to use the BatchSortAgg, we can derive the `input_order` from `LogicalAgg::o2i_col_mapping` and the `required_order` then call the `self.input().to_batch_with_order_required(input_order)`

Reply:
This is a very good point!